universal-application-tool-0.0.1/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -59,9 +73,74 @@ public ApplicantProgramBlocksController(
             httpExecutionContext.current());
   }
 
-  @DoNotCall
   public CompletionStage<Result> update(
       Request request, long applicantId, long programId, long blockId) {
-    throw new UnsupportedOperationException("Updates aren't implemented yet");
+    DynamicForm form = formFactory.form().bindFromRequest(request);
+    ImmutableMap<String, String> formData = cleanForm(form.rawData());
+
+    return applicantService
+        .stageAndUpdateIfValid(applicantId, programId, blockId, formData)
+        .thenApplyAsync(
+            (errorAndROApplicantProgramService) -> {
+              if (errorAndROApplicantProgramService.isError()) {
+                errorAndROApplicantProgramService
+                    .getErrors()
+                    .forEach(e -> logger.error("Exception while updating applicant data", e));
+                return badRequest();
```

Comment:
Will this be displayed to the user?  Probably some placeholder text here is appropriate.

Reply:
Ah, yeah. I added some text for now. Might want to spiff it up more later?